src/libcore/iter/traits/collect.rs

```diff
@@ -194,9 +194,11 @@ pub trait FromIterator<A>: Sized {
 /// `Item`:
 ///
 /// ```rust
+/// #![feature(associated_type_bounds)]
```

Comment:
Hmm; while it is unstable I think we shouldn't suggest it in such a prime piece of documentation since it will leak to stable users as well. Let's change this when ATBs are stable instead (and use `collection: impl IntoIterator<Item: Debug>` on that day).